src/_lib/modules/word-cloud/word.component.ts

```diff
@@ -0,0 +1,32 @@
+import { ChangeDetectionStrategy, Component, HostBinding, Input, OnChanges, SimpleChanges } from "@angular/core";
+import { IWord } from "./models/IWord";
+import { IWordWithPosition } from "./models/IWordWithPosition";
+
+@Component({
+    selector: 'nw-word',
+    templateUrl: './word.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class WordComponent<T extends IWord> implements OnChanges {
```

Comment:
Just out of curiosity, why do this as opposed to using IWord directly instead of T?

Reply:
The main reason for the generic type param T in the word cloud component is for the `wordsPositioned` output. The words that the user passes in go through a number of operations, like being augmented with a `fontSize` and `truncatedValue` (`IWordWithFontSize`) and then having a bounding box added in the form of `x`, `y`, `width` and `height` (`IWordWithPosition`). This is the final type that is emitted from the `wordsPositioned` output.

`@Output() wordsPositioned: EventEmitter<IWordWithPosition<T>[]> = new EventEmitter();`

Without taking `T` into consideration the user wouldn't be able to access any additional properties in the output objects that they've specified outside the constraint of `IWord`. For example, if the consumer passed in a collection of 

```
interface IMyWord extends IWord {
   id: number;
   score: number;
}
```

then they wouldn't be able to access the `id` and `score` properties on the objects emitted from the `wordsPositioned` as the output would have no knowledge of the input type

Reply:
Thanks for the info!